### PR TITLE
[wallet] Backport(0.17): Restore ability to list incoming transactions by label

### LINKED ISSUE #14382: Label support seems insufficient to replace accounts
While most of the additional functionality the old account system had over the label system were irrelevant, there is one critical functionality that no longer seems to be possible with labels - specifically, the ability to request a list of transactions (TXID + amount) that pay to addresses with a given account/label. As per the release notes for 0.17.0. the _listtransaction_ method was changed as follows;

> The account named parameter has been renamed to dummy. If provided, the dummy parameter must be set to the string *, unless running with the -deprecatedrpc=accounts argument (in which case functionality is unchanged).

_getreceivedbylabel_ and _getreceivedbyaddress_ only returns the balance, while _listreceivedbylabel_ and _listreceivedbyaddress_ do not actually filter; additionally, the former does not list transactions at all, while the latter does not include the amounts per TXID. While you could use _getaddressesbylabel_ to iterate over addresses, there is no corresponding method to get the TXIDs and amounts from this list of addresses. None of these can therefore be used as a direct replacement for this method.

The only way I can see to get anything resembling the old functionality with the API changes is to dump the entire transaction history with _listreceivedbyaddress_, filter the result by label manually, then use _gettransaction_ on each listed TXID. This is both very fiddly and very inefficient, and likely impossible when you go past a certain number of lifetime transactions.

Am I missing something here, or will it really no longer possible be to iterate over transactions received by account/label? If so, this basically means that many applications will no longer be usable with releases starting from Bitcoin Core 0.18.

### PR DESCRIPTION
**Pull request by jnewbery**
Backport of PR #14411 to v0.17.

This change partially reverts #13075 and #14023.

Fixes #14382

### CODE REVIEW

**Review by ryanofsky — APPROVED** [review]
utACK c8d1f1513fb39d80cebbc32e1b5327209485e085

File: doc/release-notes.md [hunk 1]
```diff
@@ -71,7 +71,17 @@ Notable changes
...
 =================
 
-...
+`listtransaction` label support
+-------------------------------
+
+The `listtransactions` RPC `account` parameter which was deprecated in 0.17.0
+and renamed to `dummy` has been un-deprecated and renamed again to `label`.
+
+The behavior of the `label` argument is identical to previous behavior of the
```
[the remark this answers is not in the document]
Comment (jnewbery) [reply doc/release-notes.md]: Both are good changes. Thanks!

File: doc/release-notes.md [hunk 2]
```diff
@@ -71,7 +71,17 @@ Notable changes
 0.17.x change log
 =================
 
-...
+`listtransaction` label support
```
[the remark this answers is not in the document]
Comment (jnewbery) [reply doc/release-notes.md]: fixed

File: src/wallet/rpcwallet.cpp [hunk 3]
```diff
@@ -2010,10 +2017,10 @@ UniValue listtransactions(const JSONRPCRequest& request)
     pwallet->BlockUntilSyncedToCurrentChain();
 
     std::string strAccount = "*";
-    if (!request.params[0].isNull()) {
+    if (!request.params[0].isNull() && request.params[0].get_str() != "*") {
```
[the remark this answers is not in the document]
Comment (jnewbery) [reply src/wallet/rpcwallet.cpp]: No, you're right. It's better without. The behaviour would be identical with or without the check for "*". I've removed it.

**Comment by jnewbery** [comment]
Thanks for the review @ryanofsky! I've updated the PR with your changes.

**Review by ryanofsky — COMMENTED** [review]
utACK 89306ab0df93bfdf5630910bc20b1eccb7379172. Only changes since last review were code simplification and release notes updates suggested above.

**Comment by promag** [comment]
utACK 89306ab.

**Comment by MeshCollider** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/14441/commits/89306ab0df93bfdf5630910bc20b1eccb7379172

This is slightly different from other backport PRs because the changes to `src/wallet/rpcwallet.cpp` are significantly from the original PR to achieve the same functionality. The tests are basically a clean backport though.

**Pull request merged**